Add General Guidelines for concurrent segment search (#7402)
opensearch-project/documentation-website: _search-plugins/concurrent-segment-search.md

- `0`: Use the default Lucene mechanism.
- Positive integer: Use the max target slice count mechanism. Usually, a value between 2 and 8 should be sufficient.

> ## General Guidelines

Reviewer: ```suggestion
## General guidelines
```

> Concurrent segment search helps to improve the performance of search requests at the cost of consuming more resources such as CPU, JVM, etc. It is important to try your workload to understand if the domain is sized correctly for concurrent segment search or not. We recommend the following guidelines to follow to enable concurrent segment search:

Reviewer 2: domain --> cluster

> Concurrent segment search helps to improve the performance of search requests at the cost of consuming more resources such as CPU, JVM, etc. It is important to try your workload to understand if the cluster is sized correctly for concurrent segment search or not. We recommend the following guidelines to follow to enable concurrent segment search:

Reviewer: ```suggestion
Concurrent segment search helps to improve the performance of search requests at the cost of consuming more resources such as CPU or JVM heap. It is important to test your workload in order to understand if the cluster is sized correctly for concurrent segment search. We recommend adhering to the following concurrent segment search guidelines:
```

Reviewer 3: Edited suggestion

> * Start with a slice count of 2 and measure the performance of your workload. If resource utilization is exceeding the recommended values then consider scaling your cluster. Generally based on our testing we have observed that if your workload is already consuming more than 50% of the CPU resources, then you will need to scale your cluster for concurrent segment search.

Reviewer: ```suggestion
* Start with a slice count of 2 and measure the performance of your workload. If resource utilization exceeds the recommended values, then consider scaling your cluster. Based on our testing, we have observed that if your workload is already consuming more than 50% of the CPU resources, then you need to scale your cluster for concurrent segment search.
```

Reviewer 3: Edited suggestion

> * With a slice count of 2 if you still have available resources in the cluster, then you can increase the slice count to higher number such as 4 or 6 and observe the search latency along with resource utilization on the cluster. 

Reviewer: ```suggestion
* If your slice count is 2 and you still have available resources in the cluster, then you can increase the slice count to a higher number such as 4 or 6 while monitoring search latency and resource utilization in the cluster. 
```

Reviewer 3: Edited suggestion

> * With a large number of clients sending search requests in parallel, usually a lower slice count works better. This can again be reflected in the CPU utilization, since more clients means higher TPS which will translate to higher usage of resources.

Reviewer: ```suggestion
* When many clients send search requests in parallel, usually a lower slice count works better. This is reflected in CPU utilization, because a higher number of clients leads to a higher TPS, which translates to higher resource usage.
```

Author: TPS - transactions per second. Although it probably makes sense to just explicitly say "Queries Per Second" here instead.

Reviewer 3: Edited suggestion
